### PR TITLE
feat: conditionally run pipelines in debug mode

### PR DESCRIPTION
**Pull request by YolanFery**
We want to provide a way for users to decide if they want to generate debug messages for the pipeline run. By default they will not be stored and shown.

## Changes

- Create and document a Dockerfile to create an image to easily test the local changes on a pipeline run
- Get the `log_level` settings from the environment and decide based on that if logs should be stored
- For Ruff, ignore missing docstrings in tests, I don't think they are required
- Unit test to filter the logs based on the settings

## How/what to test

- You can test changes in the SDK repo using the docker image created using the Readme instructions
- Logs messages are correctly filtered based on the settings

## Screenshots / screencast

[screen-capture (3).webm](https://github.com/user-attachments/assets/9bb2efc1-b806-41b6-9271-2b2453b82837)

## Related PRs
- https://github.com/BLSQ/openhexa-frontend/pull/908
- https://github.com/BLSQ/openhexa-app/pull/862

### CODE REVIEW

File: openhexa/sdk/pipelines/run.py [hunk 1]
```diff
@@ -76,44 +76,44 @@ def add_database_output(self, table_name: str):
...
 
     def _log_message(
         self,
-        priority: typing.Literal["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        priority: Priority,
         message: str,
     ):
-        valid_priorities = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
-        if priority not in valid_priorities:
-            raise ValueError(f"priority must be one of {', '.join(valid_priorities)}")
```
Comment (YolanFery) [inline openhexa/sdk/pipelines/run.py:104]: Not need to check with the new Typing

File: openhexa/sdk/pipelines/runtime.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def download_pipeline(url: str, token: str, run_id: str, target_dir: str):
               pipelineRun(id: $id) {
                 id
                 version {
-                  number
```
Comment (YolanFery) [inline openhexa/sdk/pipelines/runtime.py:53]: Issue also solved in https://github.com/BLSQ/openhexa-sdk-python/pull/221 but not yet merged

File: pyproject.toml [hunk 3]
```diff
@@ -67,6 +67,7 @@ include-package-data = true
 [tool.ruff]
 line-length = 120
 ignore = ["E501"]
+per-file-ignores = { "tests/**/test_*.py" = ["D100","D101","D102", "D103"] } # Ignore missing docstrings in tests
```
[the remark this answers is not in the document]
Comment (qgerome) [reply pyproject.toml:70]: I'm ok for the tests.

**Review by qgerome — COMMENTED** [review]
Looks fine for me. The `Priority` word is weird to me and would intuitively call that "LogLevel" but if you are not disturbed by this, I can live with it.

**Comment by YolanFery** [comment]
> Looks fine for me. The `Priority` word is weird to me and would intuitively call that "LogLevel" but if you are not disturbed by this, I can live with it.

I agree it looks odd but that's what we used in the GraphQL layer and it is also stored in the db (`[{"message": "Current dir /home/jovyan", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:11.094187"}, {"message": "In task 1...", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:12.995440"}, {"message": "In task 2... count is 42", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:13.469673"}]`)

So I decided to avoid complex renames and keep the same name.

I can :

1. Invest more to rename it everywhere and do the migration of existing elements in the database
2. Keep it like this
3. Rename the Enum and leave the GraphQL layer unchanged

Option 3 might be a good one, but let's agree @qgerome

**Comment by qgerome** [comment]
> > Looks fine for me. The `Priority` word is weird to me and would intuitively call that "LogLevel" but if you are not disturbed by this, I can live with it.
> 
> I agree it looks odd but that's what we used in the GraphQL layer and it is also stored in the db (`[{"message": "Current dir /home/jovyan", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:11.094187"}, {"message": "In task 1...", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:12.995440"}, {"message": "In task 2... count is 42", "**priority**": "INFO", "timestamp": "2024-12-02T15:00:13.469673"}]`)
> 
> So I decided to avoid complex renames and keep the same name.
> 
> I can :
> 
> 1. Invest more to rename it everywhere and do the migration of existing elements in the database
> 2. Keep it like this
> 3. Rename the Enum and leave the GraphQL layer unchanged
> 
> Option 3 might be a good one, but let's agree @qgerome

Let's go for option 3 then :)

**Review by qgerome — APPROVED** [review]

**Pull request merged**